firefox-ios/firefox-ios-tests/Tests/ClientTests/Microsurvey/MicrosurveyTelemetryTests.swift

```diff
@@ -0,0 +1,107 @@
+        let savedMetric = try XCTUnwrap(mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>)
+
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.surveyId, expectedSurveyId)
+        XCTAssert(savedMetric === event, "Received \(savedMetric) instead of \(event)")
+    }
+
+    func testRecordEvent_WhenDismissButtonTapped_ThenGleanIsCalled() throws {
+        let subject = createSubject()
+        let event = GleanMetrics.Microsurvey.dismissButtonTapped
+        typealias EventExtrasType = GleanMetrics.Microsurvey.DismissButtonTappedExtra
+        let expectedSurveyId = "microsurvey-id"
+
+        subject.dismissButtonTapped(surveyId: expectedSurveyId)
+
+        let savedExtras = try XCTUnwrap(mockGleanWrapper.savedExtras.first as? EventExtrasType)
+        let savedMetric = try XCTUnwrap(mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>)
+
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.surveyId, expectedSurveyId)
+        XCTAssert(savedMetric === event, "Received \(savedMetric) instead of \(event)")
+    }
+
+    func testRecordEvent_WhenUserResponseSubmitted_ThenGleanIsCalled() throws {
+        let subject = createSubject()
+        let event = GleanMetrics.Microsurvey.submitButtonTapped
+        typealias EventExtrasType = GleanMetrics.Microsurvey.SubmitButtonTappedExtra
+        let expectedSurveyId = "microsurvey-id"
+        let expectedSelection = "Neutral"
+
+        subject.userResponseSubmitted(surveyId: expectedSurveyId, userSelection: expectedSelection)
+
+        let savedExtras = try XCTUnwrap(mockGleanWrapper.savedExtras.first as? EventExtrasType)
+        let savedMetric = try XCTUnwrap(mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>)
+
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.surveyId, expectedSurveyId)
+        XCTAssertEqual(savedExtras.userSelection, expectedSelection)
+        XCTAssert(savedMetric === event, "Received \(savedMetric) instead of \(event)")
+    }
+
+    func testRecordEvent_WhenConfirmationShown_ThenGleanIsCalled() throws {
+        let subject = createSubject()
+        let event = GleanMetrics.Microsurvey.confirmationShown
+        typealias EventExtrasType = GleanMetrics.Microsurvey.ConfirmationShownExtra
+        let expectedSurveyId = "microsurvey-id"
+
+        subject.confirmationShown(surveyId: expectedSurveyId)
+
+        let savedExtras = try XCTUnwrap(mockGleanWrapper.savedExtras.first as? EventExtrasType)
+        let savedMetric = try XCTUnwrap(mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>)
+
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.surveyId, expectedSurveyId)
+        XCTAssert(savedMetric === event, "Received \(savedMetric) instead of \(event)")
```

Comment:
The change here from testing other telemetry structs with the mock is I removed the type comparison as well as the previous debug message.